src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,1323 @@
+        # '0 * numpy.arange(0, self.physicalFlybackTime, descan_period)'
+        flyback_points = numpy.zeros(math.ceil(self.physicalFlybackTime / descan_period))
+        if remainder_scanning_time != 0:
+            # Add adjusted flyback time if there is a remainder of scanning time by adding one setpoint to ensure the
+            # total scanning period is equal to a whole number of descan clock periods.
+            flyback_points = numpy.hstack((X_descan_gain * X_cell_size, flyback_points))
+
+        base = X_descan_offset - 0.5 * X_cell_size * X_descan_gain
+        setpoints = base + numpy.concatenate((scanning_points, flyback_points))
+
+        if setpoints.min() < -32768 or setpoints.max() > 32767:
+            raise ValueError("Acquisition setpoints for the descanner are out of range as excepted by the ASM API.")
+        return setpoints.astype(int).tolist()
+
+    def getYAcqSetpoints(self):
+        """
+        Creates the setpoints for the descanner in Y direction for the ASM.
+        During the scanning of a row of pixels the Y value is constant. Only one Y descan setpoint per full row of
+        pixels will be read. After completing the scan of a full row of pixels a new Y_setpoints is read which
+        describes the movement when the scanner goes from one row to the next row of pixels.
+
+        :return:  List of ints holding the setpoints in Y direction.
+        """
+        Y_scan_offset = self.scanOffset.value[1]
+        Y_scan_gain = self.scanGain.value[1]
+        Y_cell_size = self.parent._mppc.cellCompleteResolution.value[1]  # pixels
+
+        first_row_value = Y_scan_offset - 0.5 * Y_cell_size * Y_scan_gain
+        last_row_value = Y_scan_offset + 0.5 * Y_cell_size * Y_scan_gain
+        setpoints = numpy.arange(first_row_value, last_row_value, Y_scan_gain)
+
+        if len(setpoints) != Y_cell_size:
+            raise ValueError("Error in creation of Y_scan setpoints.")
+
+        if setpoints.min() < -32768 or setpoints.max() > 32767:
+            raise ValueError("Acquisition setpoints for the descanner are out of range as excepted by the ASM API.")
+        return setpoints.astype(int).tolist()
+
+
+class MPPC(model.Detector):
+    """
+    Represents the camera (mppc sensor) for acquiring the image data.
+    """
+
+    def __init__(self, name, role, parent, **kwargs):
+        """
+        Initializes the camera (mppc sensor) for acquiring the image data.
+
+        :param name(str): Name of the component
+        :param role(str): Role of the component
+        :param parent (AcquisitionServer object): Parent object of the component
+        """
+        super(MPPC, self).__init__(name, role, parent=parent, **kwargs)
+
+        # Store siblings on which this class is dependent as attributes
+        self._scanner = self.parent._ebeam_scanner
+        self._descanner = self.parent._mirror_descanner
+
+        self._shape = (8, 8)
```

Comment:
Should shape not also include the bit depth?

Reply:
Maybe I cant remember that being specified anywhere and if I quicly check the repo I can't find that anywhere else.